plugins/gzip/gzip.cc

```diff
@@ -356,36 +356,49 @@ brotli_transform_one(Data *data, const char *upstream_buffer, int64_t upstream_l
     data->bstrm.total_out = 0;
 
     data->bstrm.next_in = (uint8_t *)upstream_buffer;
-    if (!data->hc->flush()) {
-      err = BrotliEncoderCompressStream(data->bstrm.br, BROTLI_OPERATION_PROCESS, &data->bstrm.avail_in,
-                                        (const uint8_t **)&data->bstrm.next_in, &data->bstrm.avail_out, &data->bstrm.next_out,
-                                        &data->bstrm.total_out);
-    } else {
-      err = BrotliEncoderCompressStream(data->bstrm.br, BROTLI_OPERATION_FLUSH, &data->bstrm.avail_in,
-                                        (const uint8_t **)&data->bstrm.next_in, &data->bstrm.avail_out, &data->bstrm.next_out,
-                                        &data->bstrm.total_out);
-    }
 
-    if (err != BROTLI_TRUE) {
-      warning("BrotliEncoderCompressStream() call failed: %d", err);
-    }
+    ok = BrotliEncoderCompressStream(data->bstrm.br, op, &data->bstrm.avail_in, (const uint8_t **)&data->bstrm.next_in,
+                                     &data->bstrm.avail_out, &data->bstrm.next_out, &data->bstrm.total_out);
```

Comment:
Me personally I'd prefer if *ok* was type *bool* and this line was *ok = !!BrotliEncoderCompressStream(...* .

Also, using C-style casts in new C++ code is yucky.  The cleanest way to write this cast is probably:

& const_cast<const uint8_t *&>(data->bstrm.next_in)